pkg/platform/kube/apis/nuclio.io/v1beta1/types.go

```diff
@@ -43,6 +43,13 @@ func (nf *NuclioFunction) GetComputedReplicas() *int32 {
 		nf.Status.State == functionconfig.FunctionStateImported ||
 		nf.Status.State == functionconfig.FunctionStateScaledToZero ||
 		nf.Status.State == functionconfig.FunctionStateWaitingForScaleResourcesToZero {
+
+		if (nf.Status.State == functionconfig.FunctionStateWaitingForScaleResourcesToZero ||
+			nf.Status.State == functionconfig.FunctionStateScaledToZero) && nf.Spec.MinReplicas != nil && *nf.Spec.MinReplicas > 0 {
+			minReplicas := int32(*nf.Spec.MinReplicas)
+			return &minReplicas
+		}
+
```

Comment:
I think this is too late in the game for this validation / enrichment.
I would prefer it to be as early as possible - 
In the dashboard in `redeployFunction` we have the function (from `getFunction`) and the desired state. We can check there the min replicas if the desired state is `scaledToZero` and fail with `PreconditionFailed` if not supported.

Reply:
@TomerShor Currently our top logic doesn't know anything about replicas and "minReplicas" in particular. Moreover, this is logic related only to k8s. In this function we already have all the conditions associated with the replicas. I'm not sure if it's worth going higher and spreading this logic.

Reply:
We do have this knowledge in the dashboard, e.g:
```
if desiredState == functionconfig.FunctionStateWaitingForScaleResourcesToZero && 
		function.GetConfig().Spec.MinReplicas > 0 {
		return nuclio.NewErrPreconditionFailed("Cannot scale to zero a function with non-zero min replicas")
	}
```

If we validate it early, we don't need to invoke the controller at all. 
Also In general, it is better to do all validations and enrichments before we get to the controller.

Reply:
okay, will rework. But another crucial thing to note is that this change will change the behaviour since with my implementation we don't fail if desiredState is scaledToZero and minReplicas>0, we just deploy function as ready with replicas=minReplicas, but with the change we will fail and functions would not be deployed at all. If that's what we want, then fine.

Reply:
Yes that's fine, because otherwise we override the function configuration which is worse in my opinion.
It shouldn't affect the behavior with the prev-state annotation